app/realtime.php

```diff
@@ -32,41 +41,81 @@
 Swoole\Runtime::enableCoroutine(SWOOLE_HOOK_ALL);
 
-$server = new Server("0.0.0.0", 80);
+$server = new Server('0.0.0.0', 80);
+
+$server->set([
+    'websocket_compression' => true,
+    'package_max_length' => 81920
+]);
+
 $subscriptions = [];
 $connections = [];
 
-$server->on("workerStart", function ($server, $workerId) use (&$subscriptions, &$connections) {
-    Console::success('Worker '.++$workerId.' started succefully');
+$register = new Registry();
+
+$register->set('db', function () { // Register DB connection
+    $dbHost = App::getEnv('_APP_DB_HOST', '');
+    $dbUser = App::getEnv('_APP_DB_USER', '');
+    $dbPass = App::getEnv('_APP_DB_PASS', '');
+    $dbScheme = App::getEnv('_APP_DB_SCHEMA', '');
+
+    $pdo = new PDO("mysql:host={$dbHost};dbname={$dbScheme};charset=utf8mb4", $dbUser, $dbPass, array(
+        PDONative::MYSQL_ATTR_INIT_COMMAND => 'SET NAMES utf8mb4',
+        PDONative::ATTR_TIMEOUT => 3, // Seconds
+        PDONative::ATTR_PERSISTENT => true
+    ));
+
+    // Connection settings
+    $pdo->setAttribute(PDONative::ATTR_DEFAULT_FETCH_MODE, PDONative::FETCH_ASSOC);   // Return arrays
+    $pdo->setAttribute(PDONative::ATTR_ERRMODE, PDONative::ERRMODE_EXCEPTION);        // Handle all errors with exceptions
+
+    return $pdo;
+});
+
+$register->set('cache', function () { // Register cache connection
+    $redis = new Redis();
+    $redis->pconnect(App::getEnv('_APP_REDIS_HOST', ''), App::getEnv('_APP_REDIS_PORT', ''));
+    $user = App::getEnv('_APP_REDIS_USER', '');
+    $pass = App::getEnv('_APP_REDIS_PASS', '');
+    $auth = [];
+    if (!empty($user)) {
+        $auth["user"] = $user;
+    }
+    if (!empty($pass)) {
+        $auth["pass"] = $pass;
+    }
+    if (!empty($auth)) {
+        $redis->auth($auth);
+    }
+    $redis->setOption(Redis::OPT_READ_TIMEOUT, -1);
+
+    return $redis;
+});
+
+$server->on('workerStart', function ($server, $workerId) use (&$subscriptions, &$connections, &$register) {
+    Console::success('Worker ' . ++$workerId . ' started succefully');
 
     $attempts = 0;
     $start = time();
-    
+
     while ($attempts < 300) {
         try {
-            if($attempts > 0) {
-                Console::error('Pub/sub connection lost (lasted '.(time() - $start).' seconds, worker: '.$workerId.').
-                    Attempting restart in 5 seconds (attempt #'.$attempts.')');
-                sleep(5); // 1 sec delay between connection attempts
+            if ($attempts > 0) {
+                Console::error('Pub/sub connection lost (lasted ' . (time() - $start) . ' seconds, worker: ' . $workerId . ').
+                    Attempting restart in 5 seconds (attempt #' . $attempts . ')');
+                sleep(5); // 5 sec delay between connection attempts
             }
 
-            $redis = new Redis();
-            $redis->connect('redis', 6379);
-            $redis->setOption(Redis::OPT_READ_TIMEOUT, -1);
+            $redis = $register->get('cache');
```

Comment:
We should probably update the name conventions, "cache" is not the only redis job for a few versions now.